src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -508,6 +522,8 @@ internal DocumentState UpdateText(TextLoader loader, SourceText? text, Preservat
                 return documentState;
             }
 
+            Debug.Assert(text.ChecksumAlgorithm == loader.ChecksumAlgorithm);
```

Comment:
This is only called internally. TextLoader and non-null SourceText are only passed in together when the loader is created from the text. This is asserting that we have indeed created them together and the checksum algorithm flows from one to the other.

Reply:
@tmat: oh this is that stupid magic special sourceText we hold onto sometimes. Sorry I should have deleted that sooner, since I figured it was going to confuse somebody. Turns out it confused me. Got it.